.github/workflows/ci-image-scanning.yaml

```diff
@@ -5,13 +5,12 @@ on:
     # for PRs initiated by Dependabot.
     branches-ignore:
       - 'dependabot/**'
-
 permissions:
-  contents: read # for actions/checkout to fetch code
-  security-events: write # for github/codeql-action/upload-sarif to upload SARIF results
-  
+  contents: read
 jobs:
   use-trivy-to-scan-image:
+    permissions:
+      security-events: write  # for github/codeql-action/upload-sarif to upload SARIF results

```

Comment:
I'm a little hesitant about this, it will only distract we, especially since this only have one job.

Reply:
The change here is actually due to two considerations:
- Avoid top-level permissions inheriting to job-level, making job-level permissions too large. This is because the permissions needed for different jobs can vary greatly. A good practice is: setting as read-only at the [top level](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#permissions) and the required write permissions are declared at the [run-level](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idpermissions)
- refer to https://scorecard.dev/viewer/?uri=github.com/karmada-io/karmada, it's not recommended that topLevel 'security-events' permission set to 'write'.

Reply:
thanks for explain, it's make sense for me. 
/lgtm 
/approve